ikarus/utils/tensorutils.hh

```diff
@@ -150,6 +166,34 @@ auto symTwoSlots(const Eigen::TensorFixedSize<ScalarType, Eigen::Sizes<dim, dim,
   return (0.5 * (t + t.shuffle(shuffleSlot))).eval();
 }
 
+// TODO can potentially be removed or rewritten for pure tensors
+/**
+ * \brief Computes the double contraction of a 4th order tensor and a second order input tensor (here Eigen::Matrix)
+ *  \ingroup tensor
+ * \details The components of the result read  \f[ \BC_{ij} = A_{ijkl}B_{kl}  \f]
+ * \tparam ScalarType
+ * \tparam dim
+ * \param A Eigen::TensorFixedSize
+ * \param B Eigen::Matrix B
+ * \return Eigen::Matrix
+ */
+template <typename ScalarType, std::integral auto dim1, std::integral auto dim2>
+requires(dim1 == dim2)
+auto doubleContraction(const Eigen::TensorFixedSize<ScalarType, Eigen::Sizes<dim1, dim1, dim1, dim1>>& A,
+                       const Eigen::Matrix<ScalarType, dim2, dim2>& B) {
+  static constexpr int dim = dim1;
+  Eigen::Matrix<ScalarType, dim, dim> C;
+  C.setZero();
```

Comment:
This should be done with contraction: https://eigen.tuxfamily.org/dox/unsupported/eigen_tensors.html#title56

The matrix may be transformed using a tensorview. there should be some tooling in ikarus